packages/ui-react/src/components/TextField/TextField.tsx

```diff
@@ -58,6 +62,37 @@ const TextField: React.FC<Props> = ({
     className,
   );
 
+  const renderInput = () => {
+    const { required, disabled, value, ...otherInputProps } = inputProps;
+
+    // We have to explicitly check otherwise TS wont know that props.type is defined
+    const inputType = 'type' in otherInputProps ? otherInputProps.type : 'text';
```

Comment:
In which situation to we use a other type than text? Because of a rebase issue, this PR contains a lot of changes (also the case for other PRs), so it's difficult for me to see

Reply:
fixed, i've added the type as text on the component.